### PR TITLE
Minor: Move base_type to common::utils

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Closes #.

## Rationale for this change

Move this helpful function to common::utils so we can have this everywhere.

### CODE REVIEW

**Review by Weijun-H — APPROVED** [review]
LGTM! Thanks @jayzhan211 👍

File: datafusion/common/src/utils.rs [hunk 1]
```diff
@@ -390,6 +391,27 @@ pub fn arrays_into_list_array(
...
+/// Example
+/// ```
+/// use arrow::datatypes::{DataType, Field};
+/// use datafusion_common::utils::base_type;
+/// use std::sync::Arc;
+///
+/// let data_type = DataType::List(Arc::new(Field::new("item", DataType::Int32, true)));
+/// let base_type = base_type(&data_type);
+/// assert_eq!(base_type, DataType::Int32);
+pub fn base_type(data_type: &DataType) -> DataType {
```
Comment (alamb) [inline datafusion/common/src/utils.rs:405]:
```suggestion
/// ```
pub fn base_type(data_type: &DataType) -> DataType {
```

**Comment by alamb** [comment]
This PR appears to have CI test failures

**Comment by jayzhan211** [comment]
I just noticed they are different.

**Pull request closed without merging**